app/src/main/java/edu/grinnell/kdic/MainActivity.java

```diff
@@ -468,7 +468,18 @@ public void onBackPressed() {
     @Override
     protected void onDestroy() {
         Log.d(TAG, "MainActivity Destroyed.");
-
         super.onDestroy();
+        super.onDestroy();
```

Comment:
Why is this called twice? Also you want to do your clean up before you call `super.onDestroy()`